### PR TITLE
Fix `und-SA-u-ca-islamic`

### LINKED ISSUE #6559: DateTimeFormatter calendar fallback doesn't support und-SA-u-ca-islamic even though it should
`DateTimeFormatter::try_new(locale!("en-SA-u-ca-islamic").into(), ...)` should return an Umm al-Qura formatter but it doesn't.

It is a bug involving FormattableAnyCalendarKind::from_preferences.

### PR DESCRIPTION
**Pull request by robertbastian**
Fixes #6559

### CODE REVIEW

**Review by zbraniecki — APPROVED** [review]

File: components/datetime/src/scaffold/calendar.rs [hunk 1]
```diff
@@ -303,29 +303,41 @@ impl FormattableAnyCalendarKind {
...
         }
+
+        debug_assert!(false, "all locale-default calendars are supported");
+        // fall back to something non-Gregorian to make errors more obvious
+        FormattableAnyCalendarKind::Coptic
     }
 }
 
+#[test]
+fn asfd() {
```
Comment (sffc) [inline components/datetime/src/scaffold/calendar.rs]:
```suggestion
fn test_calendar_fallback() {
```

File: components/datetime/src/scaffold/calendar.rs [hunk 2]
```diff
@@ -303,29 +303,41 @@ impl FormattableAnyCalendarKind {
...
+        FormattableAnyCalendarKind::Buddhist
+    );
+    assert_eq!(
+        FormattableAnyCalendarKind::from_preferences(locale!("en-TH").into()),
+        FormattableAnyCalendarKind::Buddhist
+    );
+    assert_eq!(
+        FormattableAnyCalendarKind::from_preferences(locale!("en-SA-u-ca-islamic").into()),
+        FormattableAnyCalendarKind::HijriUmmAlQura
+    );
```
Comment (sffc) [inline components/datetime/src/scaffold/calendar.rs:338]: Nit: Add a test for `Hijri(None)` in a different region, and make it fall back to e.g. `HijriTabularTypeIIThursday`

Comment (robertbastian) [reply components/datetime/src/scaffold/calendar.rs:338]: I don't think we reached consensus on a default Hijri calendar outside of Saudi-Arabia.

**Review by sffc — APPROVED** [review]

File: components/datetime/src/scaffold/calendar.rs (hunk 2 above)
Comment (sffc) [reply components/datetime/src/scaffold/calendar.rs:338]: Ok, it's an easy thing to change now that the fallback algorithm is fixed
